ProcessLib/ComponentTransport/CreateComponentTransportProcess.cpp

```diff
@@ -0,0 +1,164 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2017, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "CreateComponentTransportProcess.h"
+
+#include "MaterialLib/Fluid/Density/CreateFluidDensityModel.h"
+#include "MaterialLib/Fluid/Viscosity/CreateViscosityModel.h"
+
+#include "ProcessLib/Parameter/ConstantParameter.h"
+#include "ProcessLib/Utils/ParseSecondaryVariables.h"
+#include "ProcessLib/Utils/ProcessUtils.h"
+
+#include "CreatePorousMediaProperties.h"
+#include "ComponentTransportProcess.h"
+#include "ComponentTransportProcessData.h"
+
+namespace ProcessLib
+{
+namespace ComponentTransport
+{
+std::unique_ptr<Process> createComponentTransportProcess(
+    MeshLib::Mesh& mesh,
+    std::unique_ptr<ProcessLib::AbstractJacobianAssembler>&& jacobian_assembler,
+    std::vector<ProcessVariable> const& variables,
+    std::vector<std::unique_ptr<ParameterBase>> const& parameters,
+    unsigned const integration_order,
+    BaseLib::ConfigTree const& config)
+{
+    //! \ogs_file_param{prj__processes__process__type}
+    config.checkConfigParameter("type", "ComponentTransport");
+
+    DBUG("Create ComponentTransportProcess.");
+
+    // Process variable.
+
+    //! \ogs_file_param{prj__processes__process__ComponentTransport__process_variables}
+    auto const pv_config = config.getConfigSubtree("process_variables");
+
+    auto process_variables = findProcessVariables(
+        variables, pv_config,
+        {
+        //! \ogs_file_param_special{prj__processes__process__ComponentTransport__process_variables__concentration}
+        "concentration",
+        //! \ogs_file_param_special{prj__processes__process__ComponentTransport__process_variables__pressure}
+        "pressure"});
+
+    auto const& porous_medium_configs =
+        //! \ogs_file_param{prj__processes__process__ComponentTransport__porous_medium}
+        config.getConfigSubtree("porous_medium");
+    PorousMediaProperties porous_media_properties{
+        createPorousMediaProperties(mesh, porous_medium_configs)};
+
+    //! \ogs_file_param{prj__processes__process__ComponentTransport__fluid}
```

Comment:
Suggest to use class FluidProperties, which contains density, viscosity, specific heat capacity and thermal conductivity (optional). There is a creator for FluidProperties,  please see
https://github.com/ufz/ogs/blob/master/ProcessLib/LiquidFlow/CreateLiquidFlowMaterialProperties.cpp#L49